components/scheduler/fx.go

```diff
@@ -23,18 +23,36 @@
 package scheduler
 
 import (
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/service/worker/scheduler"
 	"go.uber.org/fx"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/resource"
 )
 
 var Module = fx.Module(
 	"component.scheduler",
 	fx.Provide(ConfigProvider),
 	fx.Provide(TaskExecutorOptionsProvider),
+	fx.Provide(scheduler.NewSpecBuilder),
 	fx.Invoke(RegisterTaskSerializers),
 	fx.Invoke(RegisterStateMachine),
 	fx.Invoke(RegisterExecutor),
 )
 
-func TaskExecutorOptionsProvider() TaskExecutorOptions {
-	return TaskExecutorOptions{}
+func TaskExecutorOptionsProvider(
```

Comment:
nit: you can change `TaskExecutorOptions` to embed an `fx.In` and get rid of this provider.